### ISSUE TITLE
Volume mapping hangs when trying to mount a large volume > 5TB

### ISSUE THREAD

**Issue by kokhang** [rook/rook#1280]
This issue was found by @dimm0 which triggered a side effect issue which is tracked in https://github.com/rook/rook/issues/1214.

From https://github.com/rook/rook/issues/1214#issuecomment-344374292, the RBD command hangs(or take too long) when mapping the large volume. After sometime, Rook times out and issue a SIGTERM on the RBD command and try again.

**Comment by robertrbruno** [comment]
I too am seeing very long mount time for large volumes.  I created a pod with 4 volumes two fairly small and two 10TB a piece.  The pod took about 2+ hours to create and the error was mounting issues of the two large volumes.  The volumes were in a bound state very quickly.

Talking in Slack there was though it may be some of the configured time out values.  If this is the case until a full workaround is figured out can these time out values be configurable.  If this is done, I am more than happy to configure with a longer timeout and run a quick test on our system.

**Comment by dimm0** [comment]
It seems like the issue was fixed, at least for me. It does take a long time, but eventually a pod starts.

**Comment by robertrbruno** [comment]
Yes for me it does not hang just takes a long period of time.  I am trying to gauge is 2+ hours for 2 x 10TB volumes expected behavior.

**Comment by jbw976** [comment]
Can someone from the Ceph team comment on if mapping a large RBD volume is expected to take hours like is being reported? Any recommended troubleshooting steps?

/cc @travisn @rootfs @dmick

**Comment by dmick** [comment]
I've never heard of mounts taking a long time, no.  That sounds like something else is wrong.  But I guess Rook/Kubernetes will be doing more than mounting (like maybe mkfs), and maybe the mkfs is pathological somehow?...

**Comment by dimm0** [comment]
Yeah, that's most likely because of formatting
Should we try optimizing settings for xfs?
f.e. https://unix.stackexchange.com/questions/344800/mkfs-is-extremely-slow

**Comment by jbw976** [comment]
From my recollection, this timeout was from the `rbd map` attempt, before any subsequent commands such as mounting/formatting were ever attempted.  Next time I see logs for this issue I will try to confirm.  @robertrbruno did you have recent logs from the rook agent pod when you ran into this recently?  Thank you.

**Comment by robertrbruno** [comment]
Looking at the logs I do see quite a few entries that state:

"flexdriver: Ignore error about Mount failed: exit status 32. Kubernetes does this to check whether the volume has been formatted.  It will format and retry again...."

I see several (over 30 times) of those in the log from the day I was building the pod.  I see the pattern of flexdriver saying the volume has been attached and mounted, then the above error, and then another formatting volume line.  When things are finally successful I see the same volume is attached and mounted line but no error message after.  Hope this helps.

**Comment by patrickstjohn** [comment]
Running into the same issue as described above. After we have approximately 1PB in our cluster across the object store and RBD creating new volumes of anything above 1TB takes more than an hour. Rolling out any sort of large scale statefuleset then becomes essentially unusable.

**Comment by dimm0** [comment]
Btw, deleting volumes is slow too... I have a bunch of `Exceeded failedDeleteThreshold threshold: 15, for volume "pvc-0a5f0eba-6385-11e8-8d4e-0cc47a6be994", provisioner will not attempt retries for this volume` in operator logs, and I manually clean those from rbd and then remove the pv from kubernetes. One 150TB volume is already being deleted manually for more than a day, and so far it's deleted for 33%

**Comment by patrickstjohn** [comment]
Potential workaround: change the default filesystem from ext4 to xfs in your StorageClass. The timeouts don't happen and our build goes from 2-3hrs down to 15minutes. Additionally, it looks like Ceph recommends xfs as your default filesystem format so that probably should be the sane default.

```yaml
fstype: xfs
```

**Comment by jbw976** [comment]
@patrickstjohn would you be interested in submitting a PR that updates the default to xfs? :sunglasses:

**Comment by patrickstjohn** [comment]
@jbw976 absolutely happy to. It looks like the default is being pulled in from Kubernetes instead of being explicitly set by Rook. https://github.com/rook/rook/pull/847 and https://github.com/kubernetes/kubernetes/blob/master/pkg/util/mount/mount_linux.go#L472

I updated the example and the documentation but an override of Kubernetes defaults is probably a larger discussion for your team.

**Comment by dimm0** [comment]
xfs is not helping the issue, we're using it by default

**Comment by jbw976** [comment]
Thanks for your input @dimm0, that would imply that the `rbd map` command itself that performs the map/attach does take a significant amount of time.
